### DISCUSSION TITLE
What stream does music mode download?

### DISCUSSION THREAD

**Discussion by ballo** [Thunderarea/music-mode-for-youtube#14] [General]
For some reason when switching from Music mode to normal mode, there is no delay in the video. That seems highly counter-intuitive as switching between the audio-only stream and the video stream on the app incurs a delay as the video track is loaded. It's almost like this extension is loading the video stream in the music mode.

**Comment by Thunderarea** [comment]
Hi, when the extension is enabled it downloads only the audio stream and the respective video stream is completely blocked (except for live videos where the video is not blocked). When you are switching to video there are some reasons that determine whether the video will load quickly or not. First of all, if the video is still in the start it will load quickly because it will have to load only the first seconds. Secondly, the time that YouTube will need to load the video track it depends a lot on the file size of the video and the resolution. Videos with low resolution will load faster. Videos that just show an image will load faster than respective videos that contain video. There are many reasons that determine the size of the video. Two videos with the same length and resolution can have big differences in the size.  

However, in order to be sure that this is not caused by a bug you can check if the extension blocks the video by following these steps:

1. Go to YouTube
1. Disable the extension
1. Open the Developer Tools. You can do it with one of the following ways:
    - Right click on the page and then click "Inspect"
    - Press F12
    - Press Ctrl + Shift + I
    - Go to Chrome menu → More Tools → Developer Tools
1. In the Developer Tools go to the "Network" tab. if you cannot see the tab click on the ">>" icon and select the "Network" tab
6. In the Filter bar write "mime=video"
7. Clear the tab by using the second icon from the left in the second row
8. Click on a video to play
9. You will see the video requests in the Network tab (requests to download video file)
10. Now, play a video with the extension enabled. There will be no new video requests but only audio requests (you can see the audio requests by writing in the Filter bar "mime=audio")

**Comment by ballo** [comment]
I did some tests forcing a slower connections and it behaves a bit more as expected. Thanks for the reply.
